db/inmemory_test.go

```diff
@@ -0,0 +1,54 @@
+package db
+
+import "testing"
+
+func TestBasic(t *testing.T) {
+	im := NewInMemoryVolumeDatabase()
+
+	err := im.Connect()
+	if err != nil {
+		t.Fatal("Connect returned an error! " + err.Error())
+	}
+
+	list, err := im.List()
+	if err != nil {
+		t.Fatal("List returned an error! " + err.Error())
```

Comment:
This marks the test as a failure and logs, but does not prevent the later checks from running.

Specifically, lines 22-53 would also have a chance to log their errors. (the significance of this I explained in my "Request Changes" comment)